### PR TITLE
FieldValues: Implement array accessors for deprecated Vector types

### PR DESCRIPTION
**Pull request by ryantxu**
See https://github.com/grafana/grafana/issues/66480

This PR wraps the few remaining Vector implementations with an ES6 proxy that supports array index lookups.

### CODE REVIEW

File: packages/grafana-data/src/vector/CircularVector.ts [hunk 1]
```diff
@@ -34,6 +36,7 @@ export class CircularVector<T = any> extends FunctionalVector<T> {
     if (options.capacity) {
       this.setCapacity(options.capacity);
     }
+    return makeArrayIndexableVector(this);
```
Comment (ryantxu) [inline packages/grafana-data/src/vector/CircularVector.ts:39]: this is the only version that is still in (limited) real use -- the rest can be replaced by arrays easily.

File: packages/grafana-data/src/vector/ConstantVector.test.ts [hunk 2]
```diff
@@ -10,8 +10,9 @@ describe('ConstantVector', () => {
     expect(v.get(1)).toEqual(value);
 
     // Now check all of them
-    for (let i = 0; i < 10; i++) {
+    for (let i = 0; i < 7; i++) {
```
Comment (ryantxu) [inline packages/grafana-data/src/vector/ConstantVector.test.ts:13]: previously the length was irrelevant, now it is real

File: packages/grafana-data/src/types/vector.ts [hunk 3]
```diff
@@ -97,3 +97,30 @@ export interface ReadWriteVector<T = any> extends Vector<T> {}
  * @deprecated -- this is now part of the base Vector interface
  */
 export interface MutableVector<T = any> extends ReadWriteVector<T> {}
+
+/**
+ * This is an extreemly inefficient Vector wrapper that allows vecorts to
```
Comment (nmarrs) [inline packages/grafana-data/src/types/vector.ts:102]:
```suggestion
 * This is an extreemly inefficient Vector wrapper that allows vectors to
```

File: packages/grafana-data/src/types/vector.ts [hunk 4]
```diff
@@ -97,3 +97,30 @@ export interface ReadWriteVector<T = any> extends Vector<T> {}
...
+ * be treated as arrays.  We should avoid using this wrapper, but it is helpful
+ * for a clean migration to arrys
```
Comment (nmarrs) [inline packages/grafana-data/src/types/vector.ts:104]:
```suggestion
 * for a clean migration to arrays
```

**Review by nmarrs — APPROVED** [review]

**Pull request merged**